composer/loggers/tensorboard_logger.py

```diff
@@ -75,20 +79,34 @@ def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
                 pass
 
     def init(self, state: State, logger: Logger) -> None:
-        from torch.utils.tensorboard import SummaryWriter
+        self.run_name = state.run_name
 
         # We set the log_dir to a constant, so all runs can be co-located together.
         if self.log_dir is None:
             self.log_dir = 'tensorboard_logs'
 
+        self._initialize_summary_writer()
+
+    def _initialize_summary_writer(self):
+        from torch.utils.tensorboard import SummaryWriter
+
+        assert self.run_name is not None
+        assert self.log_dir is not None
         # We name the child directory after the run_name to ensure the run_name shows up
         # in the Tensorboard GUI.
-        summary_writer_log_dir = Path(self.log_dir) / state.run_name
+        summary_writer_log_dir = Path(self.log_dir) / self.run_name
 
         # To disable automatic flushing we set flushing to once a year ;)
         flush_secs = 365 * 3600 * 24
         self.writer = SummaryWriter(log_dir=summary_writer_log_dir, flush_secs=flush_secs)
 
+        if self.event_file_base_file_path is None:
+            self.event_file_base_file_path = self.writer.file_writer.event_writer._file_name
+
+        # Give event_file a unique name to avoid appending to the same file on every flush.
+        self.writer.file_writer.event_writer._file_name = self.event_file_base_file_path + f'-{self.flush_count}'
```

Comment:
wouldn't a unique filename for every flush lead to creation a ton of files / storage consumption for a single run? If the observed throughput degradation could be fixed by increasing the flush_interval, is that not sufficient?

I'm also not thrilled about accessing private fields of the `SummaryWriter` in this way.

Reply:
using the Hanlin's `close()` idea instead.